lib/ansible/modules/remote_management/redfish/redfish_config.py

```diff
@@ -0,0 +1,233 @@
+      bios_attr_value: Bios
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Enable PXE Boot for NIC1
+    redfish_config:
+      category: Systems
+      command: SetBiosAttributes
+      bios_attr_name: PxeDev1EnDis
+      bios_attr_value: Enabled
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Set BIOS default settings
+    redfish_config:
+      category: Systems
+      command: SetBiosDefaultSettings
+      baseuri: "{{ baseuri }}"
+      user: "{{ user }}"
+      password: "{{ password }}"
+
+  - name: Enable NTP in the OOB Controller
+    redfish_config:
+      category: Manager
+      command: SetManagerAttributes
+      mgr_attr_name: NTPConfigGroup.1.NTPEnable
+      mgr_attr_value: Enabled
+      baseuri: "{{ baseuri }}"
+      user: "{{ user}}"
+      password: "{{ password }}"
+
+  - name: Set NTP server 1 to {{ ntpserver1 }} in the OOB Controller
+    redfish_config:
+      category: Manager
+      command: SetManagerAttributes
+      mgr_attr_name: NTPConfigGroup.1.NTP1
+      mgr_attr_value: "{{ ntpserver1 }}"
+      baseuri: "{{ baseuri }}"
+      user: "{{ user}}"
+      password: "{{ password }}"
+
+  - name: Set Timezone to {{ timezone }} in the OOB Controller
+    redfish_config:
+      category: Manager
+      command: SetManagerAttributes
+      mgr_attr_name: Time.1.Timezone
+      mgr_attr_value: "{{ timezone }}"
+      baseuri: "{{ baseuri }}"
+      user: "{{ user}}"
+      password: "{{ password }}"
+'''
+
+RETURN = '''
+result:
+    description: different results depending on task
+    returned: always
+    type: dict
+    sample: BIOS Attributes set as pending values
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.redfish_utils import RedfishUtils
+from ansible.module_utils._text import to_native
+
+
+# More will be added as module features are expanded
+CATEGORY_COMMANDS_ALL = {
+    "Systems": ["SetBiosDefaultSettings", "SetBiosAttributes"],
+    "Manager": ["SetManagerAttributes"],
+}
+
+
+def main():
+    result = {}
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True),
+            command=dict(required=True, type='list'),
+            baseuri=dict(required=True),
+            user=dict(required=True),
+            password=dict(required=True, no_log=True),
+            mgr_attr_name=dict(),
+            mgr_attr_value=dict(),
+            bios_attr_name=dict(),
+            bios_attr_value=dict(),
+        ),
+        supports_check_mode=False
+    )
+
+    category = module.params['category']
+    command_list = module.params['command']
+
+    # admin credentials used for authentication
+    creds = {'user': module.params['user'],
+             'pswd': module.params['password']}
+
+    # Manager attributes to update
+    mgr_attributes = {'mgr_attr_name': module.params['mgr_attr_name'],
```

Comment:
The parameters `mgr_attr_value` and `bios_attr_name` are not marked as required nor do they have a default value in the argument spec. This means they will be `None` if not passed into the module. Is that ok?

Reply:
Good catch. Need to set to a bogus default value, as setting to a valid default value can have unintended consequences. With an 'invalid' value, it will catch the error and exit.
```
  mgr_attr_name=dict(default='invalid'),
  mgr_attr_value=dict(default='invalid'),
  bios_attr_name=dict(default='invalid'),
  bios_attr_value=dict(default='invalid'),
```
Without a default value, and if not specified, error message is ugly.